py/mpstate.h

```diff
@@ -242,6 +242,8 @@ typedef struct _mp_state_thread_t {
     mp_obj_dict_t *dict_locals;
     mp_obj_dict_t *dict_globals;
 
+    mp_obj_base_t *cur_exc;
```

Comment:
So, we already have `cur_exception`. But it's currently in mp_state_vm_t, which is not correct: https://docs.python.org/3/library/sys.html#sys.exc_info

> The information returned is specific both to the current thread and to the current stack frame.

Please move it to mp_state_thread_t and apparently use for this patchset.

Reply:
I added a new entry to not tangle this set of patches too much with the existing code.  The idea would be to clean up things like this if the patches get merged.  And it's not immediately clear that `cur_exc` and `cur_exception` can be the same thing, due to the VM setting/clearing `cur_exception` at various points.